### ISSUE TITLE
Production pods

### ISSUE THREAD

**Issue by scothis** [spring-cloud-incubator/mononoke#2]
What does a Pod running a Spring Boot application look like?

- how are resource requests and limits set?
- how are lifecycle probes defined?
- how is shutdown managed?
- what environment variables are set?
- how are ports allocated?
- ...?

We should collect pod yaml in this issue that shows the current best practices for running a boot app on k8s.

**Comment by ojhughes** [comment]
Using the manifests here as a starting point but they are missing a couple of bits I think 
https://github.com/spring-cloud/spring-cloud-dataflow/tree/master/src

**Comment by ojhughes** [comment]
See 3b5f2cd722f8b643c0773b673ae9bb76ece1709e for the base that will be used to define the Pod yaml

**Comment by scothis** [comment]
What does an app deployed by SCDF look like?

**Comment by ojhughes** [comment]
The manifests are for the [data flow server](https://docs.spring.io/autorepo/docs/spring-cloud-dataflow-server-kubernetes-docs/current/reference/htmlsingle/), which is also a Spring Boot application.

The data flow workloads are created using[ the spring cloud deployer framework](https://docs.spring.io/autorepo/docs/spring-cloud-dataflow-server-kubernetes-docs/current/reference/htmlsingle/#_application_and_server_properties)

Will materialize an example manifest YAML from a deployed task

**Comment by scothis** [comment]
@ojhughes what we need to know isn't what knobs are available, or even how SCDF exposes those knobs. We need to collect what meaningful semantic settings for the knobs are, and how they vary between use cases.

**Comment by ojhughes** [comment]
@scothis yes understood, so a vanilla Spring Boot webmvc deployment with production configuration. The dataflow server manifests are a good starting point to build this,  as itis quite a battle tested configuration

**Comment by ojhughes** [comment]
Closed by mistake
